app/Http/Middleware/SubscribersConfigured.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of Cachet.
+ *
+ * (c) Cachet HQ <support@cachethq.io>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CachetHQ\Cachet\Http\Middleware;
+
+use CachetHQ\Cachet\Facades\Setting;
+use Closure;
+use Illuminate\Support\Facades\Redirect;
+
+class SubscribersConfigured
+{
+    /**
+     * We're verifying that subscribers is both enabled and configured.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure                 $next
+     *
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        $isEnabled = Setting::get('enable_subscribers', false);
+        $mailAddress = env('MAIL_ADDRESS', false);
+        $mailFrom = env('MAIL_NAME', false);
+
+        if (!($isEnabled && $mailAddress && $mailFrom)) {
+            return Redirect::route('status-page');
```

Comment:
I was thinking about this as I was doing it... In most of our existing cases, we should show an error page.

With this though, I don't want to *slow down* access to a status page, just because a feature isn't enabled. I don't know, to be honest I'm not fussed either way and I agree that we're doing too many redirects like this.

If we do start going to an error page, we need to make a nice one at least.